pkg/clusteragent/orchestrator/transformer.go

```diff
@@ -194,3 +197,173 @@ func extractDeploymentConditionMessage(conditions []v1.DeploymentCondition) stri
 	}
 	return ""
 }
+
+func extractNode(n *corev1.Node) *model.Node {
+	msg := &model.Node{
+		Metadata:      orchestrator.ExtractMetadata(&n.ObjectMeta),
+		PodCIDR:       n.Spec.PodCIDR,
+		PodCIDRs:      n.Spec.PodCIDRs,
+		ProviderID:    n.Spec.ProviderID,
+		Unschedulable: n.Spec.Unschedulable,
+		Status: &model.NodeStatus{
+			Allocatable: map[string]int64{},
+			Capacity:    map[string]int64{},
+		},
+	}
+
+	if len(n.Spec.Taints) > 0 {
+		msg.Taints = extractTaints(n.Spec.Taints)
+	}
+
+	// status
+	msg.Status.KubeletVersion = n.Status.NodeInfo.KubeletVersion
+	msg.Status.ContainerRuntimeVersion = n.Status.NodeInfo.ContainerRuntimeVersion
+	msg.Status.OperatingSystem = n.Status.NodeInfo.OperatingSystem
+	msg.Status.Architecture = n.Status.NodeInfo.Architecture
+	msg.Status.OsImage = n.Status.NodeInfo.OSImage
+	msg.Status.KubeProxyVersion = n.Status.NodeInfo.KubeProxyVersion
+	msg.Status.KernelVersion = n.Status.NodeInfo.KernelVersion
+	extractCapacitiesAndAllocatables(n, msg)
+
+	// extract status addresses
+	if len(n.Status.Addresses) > 0 {
+		msg.Status.NodeAddresses = map[string]string{}
+		for _, address := range n.Status.Addresses {
+			msg.Status.NodeAddresses[string(address.Type)] = address.Address
+		}
+	}
+
+	// extract conditions
+	if len(n.Status.Conditions) > 0 {
```

Comment:
the range won't do a thing if `len == 0`, so I guess it could be removed?

Reply:
Makes sense. Forgot that `len of nil` == 0 😳